aws/sdk/integration-tests/ec2/tests/paginators.rs

```diff
@@ -50,3 +50,39 @@ async fn paginators_handle_empty_tokens() {
     assert_eq!(first_item, None);
     conn.assert_requests_match(&[]);
 }
+
+/// See https://github.com/awslabs/aws-sdk-rust/issues/405
+///
+/// EC2 can also reply with the token truly unset which will be interpreted as `None`
```

Comment:
What exactly does this mean? How many "kinds" of unset are there? I can think of
a. field returned but empty or set to a value signifying empty
b. no field returned

Were we only covering case a previously and now we're covering case b too?

Reply:
yeah it can either be `<nextToken></nextToken>`, or as is the case here, `<nextToken>` is not present in the response at all. In one case, we get `Some("")`, in the other, we get `None`. `None` used to be handled properly, but regressed when I added the failsafe to avoid infinite loops